packages/jazz-tools/src/coValues/deepLoading.ts

```diff
@@ -145,7 +149,7 @@ export type DeeplyLoaded<
   CurrentDepth extends number[] = [],
 > = DepthLimit extends CurrentDepth["length"]
   ? V
-  : Depth extends true | undefined
+  : Depth extends boolean | undefined // Checking against boolean instead of true because the inference from RefsToResolveStrict transforms true into boolean
```

Comment:
Applying any kind of `extends` check on the generic R makes Typescript change the inference of `true` to boolean.

![Screenshot 2025-02-10 at 19 53 27](https://github.com/user-attachments/assets/41da4fb3-b361-45d6-8d19-03a4c098ec24)

Had to change this here to avoid getting broken types from Resolved:
![Screenshot 2025-02-10 at 19 06 05](https://github.com/user-attachments/assets/b4f32dfc-10cc-4074-99f1-7e383526a2de)

Since the type becomes { $each: boolean } the if we don't stop the `false` types here the `DeeplyLoaded` type would continue the recursion until reaching the limit of 5.